### PR TITLE
Fixing Ticket Request CSV as per request; upgrading gems

### LINKED ISSUE #185: Ticket Request CSV
**Is your feature request related to a problem? Please describe.**
Tickets would like the Ticket Request CSV reverted back to its original format which includes column order. The CSV is imported as a spreadsheet and is the basis for downstream spreadsheets that help Tickets validate users and run analytics. 

**Describe the solution you'd like**
We would like the original format restored which includes the original ordering of the columns. Below is a mapping of the original columns to the new columns. 

Each record in the Ticket Request CSV should represent a Ticket Request. It should not represent a Guest/Ticket holder/Attendee.

| **Original columns**        | **New columns**             |
| ----------------------- | ----------------------- |
| name                    | Name                    |
| email                   | Email                   |
| id                      |                         |
| adults                  | Adults                  |
| kids                    | Kids                    |
| cabins                  |                         |
| needs_assistance        | Needs Assistance        |
| notes                   | Notes                   |
| status                  | Status                  |
| created_at              |                         |
| updated_at              |                         |
| user_id                 |                         |
| special_price           | Special Price           |
| event_id                |                         |
| donation                | Donation                |
| role                    | Role                    |
| role_explanation        | Role Explanation        |
| previous_contribution   | Previous Contribution   |
| address_line1           | Address Line1           |
| address_line2           | Address Line2           |
| city                    | City                    |
| state                   | State                   |
| zip_code                | Zip Code                |
| country_code            | Country Code            |
| admin_notes             | Admin Notes             |
| car_camping             | Car Camping             |
| car_camping_explanation | Car Camping Explanation |
| agrees_to_terms         |                         |
| early_arrival_passes    | Early Arrival Passes    |
| late_departure_passes   | Late Departure Passes   |
| guests                  |                         |
|                         | Guest?                  |
|                         | Kids Age                |

Example CSV values of the original format.
```
name,email,id,adults,kids,cabins,needs_assistance,notes,status,created_at,updated_at,user_id,special_price,event_id,donation,role,role_explanation,previous_contribution,address_line1,address_line2,city,state,zip_code,country_code,admin_notes,car_camping,car_camping_explanation,agrees_to_terms,early_arrival_passes,late_departure_passes,guests
Michael Matera,spamme@fatboycentral.com,3018,2,0,0,false,I love you all.,C,2024-05-13 21:30:30 -0700,2024-05-13 21:34:31 -0700,37,,14,0.0,contributor,I do tech stuff!,Coordinator,,,,,,,,,,true,0,0,"[""Michael Matera <spamme@fatboycentral.com>"", ""Some Person <some@person.net>""]"
Yarun Luon,yarun@fnf.org,3019,3,0,0,false,,C,2024-05-13 21:34:18 -0700,2024-05-13 21:43:00 -0700,292,,14,0.0,coordinator,tickets,tickets,,,,,,,,,,true,0,0,"[""Yarun Luon <yarunl@gmail.com>"", ""Heidi"", ""Tyler""]"
Matt Levy,matt@fnf.org,3020,4,0,0,false,,C,2024-05-13 21:40:21 -0700,2024-05-13 21:43:19 -0700,1281,,14,0.0,uber_coordinator,Raver Yoda,Play,,,,,,,,,,true,0,0,"[""Matt Levy <matt@fnf.org>"", ""Laina Levy <lainalevy@gmail.com"", ""Oliver Levy <octodolphin@gmail.com>"", ""Kathy Miller""]"
Jane Jach,janejach@gmail.com,3109,2,0,0,false,paying for Fil's ticket now; Jane will pay via PayPal week-of event in case she's unable to go (taking care of mom),C,2024-05-18 11:54:51 -0700,2024-05-18 12:03:26 -0700,637,220.0,14,0.0,coordinator,Jane - tickets; Fil - skipper,same,,,,,,,,,,true,0,0,"[""Jane Jach <janejach@gmail.com>"", ""Fil Jach <filipjach@gmail.com>""]"
```

**Alternative solutions**
If the original format cannot be restored, please add back the `created_at`, `updated_at`, `id` columns. These are used for analytics and tracking. 

**Additional context**
The Ticket Request CSV is not designed to be read by humans and is meant to imported to spreadsheets for further post-processing.

### PR DESCRIPTION
**Pull request by kigster**
# Description 

This PR fixes #185 and also removes the N+1 user fetch by including it in the query.

The resulting CSV now looks as follows: 

[14-fnf-2024-summer-campout-ticket-requests-csv.csv](https://github.com/fnf-org/TicketBooth/files/15440393/14-fnf-2024-summer-campout-ticket-requests-csv.csv)

### CODE REVIEW

File: app/models/ticket_request.rb [hunk 1]
```diff
@@ -205,8 +188,9 @@ def kids_age(string)
   scope :declined, -> { where(status: STATUS_DECLINED) }
   scope :not_declined, -> { where.not(status: STATUS_DECLINED) }
 
-  # Those TicketRequests that should be exported as a Guest List
-  scope :active, -> { where(status: [STATUS_COMPLETED, STATUS_AWAITING_PAYMENT]) }
+  # Those TicketRequests that should be exported as a Guest List and include user record to avoid N+1
+  # @see https://medium.com/doctolib/how-to-find-fix-and-prevent-n-1-queries-on-rails-6b30d9cfbbaf
+  scope :active, -> { where(status: [STATUS_COMPLETED, STATUS_AWAITING_PAYMENT]).includes(:user) }
```
Comment (yarunluon) [inline app/models/ticket_request.rb:193]: Please print all statuses.

File: app/models/ticket_request.rb [hunk 2]
```diff
@@ -47,76 +47,59 @@ class << self
     #   and then for each ticket request we return the primary user with the ticket request info,
     #   followed by one row per guest.
     def for_csv(event)
-      table = []
+      [].tap do |table|
+        event.ticket_requests.active.each do |ticket_request|
```
Comment (yarunluon) [inline app/models/ticket_request.rb:51]: I don’t know what an active or an inactive table is, but the CSV should essentially be a dump of the table.

Comment (kigster) [reply app/models/ticket_request.rb:51]: Active means either paid or awaiting payment. It does not include declined, refunded etc

**Review by yarunluon — COMMENTED** [review]
Thanks for making these changes. Ticket Request CSV should print all records from the table and not filter for any records.

**Review by beingmattlevy — APPROVED** [review]

**Pull request merged**